CMakeLists.txt

```diff
@@ -28,7 +28,8 @@ ENDIF (NOT CLANG_URL)
 set(CMAKE_MACOSX_RPATH 1)
 
 IF (NOT NUM_BUILD_THREADS)
-  SET(NUM_BUILD_THREADS 8)
+  include(ProcessorCount)
+  ProcessorCount(NUM_BUILD_THREADS)
```

Comment:
Yea, it already is two separate variables. The `NUM_BUILD_THREADS` does not affect the threads for testing. Right now, I just called the variable `N`, but it does make sense to rename it to `NUM_TEST_THREADS`.